addons/mail/static/src/core/web/mail_composer_template_selector.js

```diff
@@ -0,0 +1,139 @@
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { sprintf } from "@web/core/utils/strings";
+import { standardFieldProps } from "@web/views/fields/standard_field_props";
+import { useService } from "@web/core/utils/hooks";
+
+import { Component, useState, onWillStart } from "@odoo/owl";
+import { ConfirmationDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+import { Dropdown } from "@web/core/dropdown/dropdown";
+import { DropdownItem } from "@web/core/dropdown/dropdown_item";
+import { SelectCreateDialog } from "@web/views/view_dialogs/select_create_dialog";
+
+
+export class MailComposerTemplateSelector extends Component {
+    static template = "mail.MailComposerTemplateSelector";
+    static components = { Dropdown, DropdownItem };
+    static props = { ...standardFieldProps };
+
+    setup() {
+        this.action = useService("action");
+        this.orm = useService("orm");
+        this.state = useState({});
+        this.limit = 5;
+
+        onWillStart(() => {
+            this.fetchTemplates();
+        });
+    }
+
+    async fetchTemplates() {
+        const domain = [["model", "=", this.props.record.data.render_model]];
+        const fields = ["display_name"];
+        this.state.templates = await this.orm.searchRead(
+            "mail.template",
+            domain,
+            fields,
+            { limit: this.limit }
+        );
```

Comment:
Screens have more than 20 characters available, unless we are using a calculette ?
```suggestion
        this.state.templates = await this.orm.searchRead("mail.template", domain, fields, { limit: this.limit });
```